text/0187-data-classification.md

```diff
@@ -0,0 +1,323 @@
+message Span {
+// ... omitted 
++  // classification represents a bitwise flag 
++  // that is used to represent span classification type
++  int64    classification  = 16;
+}
+
+// Defined at https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto#L113
+message LogRecord {
+// ... omitted 
++  // classification represents a bitwise flag 
++  // that is used to represent LogRecord classification type
++  int64    classification  = 11;
+}
+```
+
+Once that has been update in the protobuf definition has been published, a classification list will need to be defined as part of the SDK with space for vendors to allow for their own classification definitions.
+
+A classification MAY contain multiple classification, therefore a classification value MUST not overlap with other classification so that they can be combined together using bitwise inclusive or. The SDK will reserve the 32 Least Significant Bits (LSB) to allow for future additions that are yet to be considered; allow vendors to define values for the 32 Most Significant bits to define as they see fit.
+
+A staring list can look like the following :
+
+| Classification                     | Purpose                                                                                               | BitShift  | Hint Value (base 16)  | bit mask             |
+|----------------------------------- |------------------------------------------------------------------------------------------------------ |---------- |---------------------- |--------------------- |
+| No Value                           | Is the default value when un set                                                                      | 0 << 0    | 0x0000                | 0000 0000 0000 0000  |
+| Ephemeral                          | The attributes that are short lived and have high potential to change over extended periods of time.  | 1 << 0    | 0x0001                | 0000 0000 0000 0001  |
+| High Cardinality                   | The value is an unbounded set                                                                         | 1 << 1    | 0x0002                | 0000 0000 0000 0010  |
+| Sensitive Value                    | The value MAY contain information that requires sanitisation                                          | 1 << 2    | 0x0004                | 0000 0000 0000 0100  |
+| Personal Identifiable Information  | The value DOES contain Personal Identifiable Information                                              | 1 << 3    | 0x0008                | 0000 0000 0000 1000  |
+| User Generated Content             | The value DOES contain User Generated Content                                                         | 1 << 4    | 0x0010                | 0000 0000 0001 0000  |
+| Service Level Objective            | The value is used to track a service level objective                                                  | 1 << 5    | 0x0020                | 0000 0000 0010 0000  |
```

Comment:
I'm not sure I would include this one, at least not for the first iteration. My impression is that this wouldn't have the same audience as the other values.

Reply:
For Service Level Object? Or classifications themselves?

Reply:
I mean only the SLO classification.